services/src/test/java/edu/unc/lib/dl/cdr/services/rest/modify/ExportCsvIT.java

```diff
@@ -0,0 +1,482 @@
+import edu.unc.lib.dl.persist.services.delete.MarkForDeletionJob;
+import edu.unc.lib.dl.persist.services.edit.UpdateDescriptionService;
+import edu.unc.lib.dl.search.solr.service.ChildrenCountService;
+import edu.unc.lib.dl.search.solr.service.SolrSearchService;
+import edu.unc.lib.dl.sparql.FedoraSparqlUpdateService;
+import edu.unc.lib.dl.test.AclModelBuilder;
+import edu.unc.lib.dl.test.RepositoryObjectTreeIndexer;
+import edu.unc.lib.dl.util.ResourceType;
+
+/**
+ *
+ * @author lfarrell
+ *
+ */
+@ContextHierarchy({
+        @ContextConfiguration("/spring-test/test-fedora-container.xml"),
+        @ContextConfiguration("/spring-test/cdr-client-container.xml"),
+        @ContextConfiguration("/spring-test/solr-indexing-context.xml"),
+        @ContextConfiguration("/export-csv-it-servlet.xml")
+})
+public class ExportCsvIT extends AbstractAPIIT {
+    private static final String COLLECTION_PATH =
+            "/Content Collections Root/Admin unit/Collection";
+    private static final String FOLDER_PATH =
+            "/Content Collections Root/Admin unit/Collection/Folder";
+    private static final Path MODS_PATH_1 = Paths.get("src/test/resources/mods/valid-mods.xml");
+    private static final Path MODS_PATH_2 = Paths.get("src/test/resources/mods/work-mods.xml");
+
+    @Autowired
+    protected String baseAddress;
+    @Autowired
+    protected File solrDataDir;
+    @Autowired
+    protected EmbeddedSolrServer server;
+    @Autowired
+    protected ChildrenCountService childrenCountService;
+    @Autowired
+    protected SolrUpdateDriver driver;
+    @Autowired
+    protected SolrSearchService solrSearchService;
+    @Resource(name = "accessGroups")
+    protected AccessGroupSet accessGroups;
+    @Autowired
+    protected Model queryModel;
+    @Autowired
+    protected RepositoryObjectLoader repositoryObjectLoader;
+    @Autowired
+    protected RepositoryObjectFactory repositoryObjectFactory;
+    @Autowired
+    protected DocumentIndexingPackageFactory dipFactory;
+    @Autowired
+    protected RepositoryPIDMinter pidMinter;
+    @Autowired
+    private RepositoryInitializer repoInitializer;
+    @Autowired
+    private RepositoryObjectTreeIndexer treeIndexer;
+    @Autowired
+    private RepositoryObjectSolrIndexer solrIndexer;
+    @Autowired
+    private FedoraSparqlUpdateService sparqlUpdateService;
+    @Autowired
+    private UpdateDescriptionService updateDescService;
+
+    protected ContentRootObject rootObj;
+    protected AdminUnit unitObj;
+    protected CollectionObject collObj;
+    protected CollectionObject collObj2;
+    protected FolderObject folderObj;
+
+    @Before
+    public void setup() throws Exception {
+        setupContentRoot();
+        generateBaseStructure();
+
+        treeIndexer.indexAll(baseAddress);
+
+        setField(solrSearchService, "solrClient", server);
+        setField(childrenCountService, "solrClient", server);
+
+        solrIndexer.index(rootObj.getPid(),
```

Comment:
One other note, this test class takes a rather long time, on my local machine it is taking 40-60 seconds. Some of that is the cost of starting up fedora and solr, which may be unavoidable. But I think the test can be a good bit shorter if we reduce the number of calls to treeIndexer and solrIndexer, since some of those are taking a few seconds per call. For the solrIndexer in particular, it should be okay to wait until everything has been set up and then call it, rather than call it in the setup method and then once or twice more in the body. I'm happy to help out with this after other changes have been made if you'd prefer.

Additionally, creating fewer fedora objects may help some. For example, in `addFolderAndWork` you could add works directly to `folderObj` rather than creating a new folder

At some point we could probably make `treeIndexer.indexAll` more efficient by pointing at specific objects rather than the content root, which might make the test suite faster over all, but we can do that another time.